src/meta/meta_turtle.c

```diff
@@ -81,12 +93,81 @@ __metadata_load_hot_backup(WT_SESSION_IMPL *session)
         WT_ERR(__wt_getline(session, fs, value));
         if (value->size == 0)
             WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
+        /*
+         * When performing partial backup restore, generate a list of tables that is not part of the
+         * target uri list so that we can drop all entries later. To do this, parse through all the
+         * table metadata entries and check if the metadata entry exists in the target uri hash
+         * table. If the metadata entry doesn't exist in the hash table, append the table name to
+         * the partial backup list.
+         */
+        metadata_key = (char *)key->data;
+        if (F_ISSET(conn, WT_CONN_BACKUP_PARTIAL_RESTORE) &&
+          WT_PREFIX_MATCH(metadata_key, "table:"))
+            /*
+             * The target uri will be the deciding factor if a specific metadata table entry needs
+             * to be dropped. If the metadata table entry does not in the target uri hash table,
+             * append the metadata key to the backup remove list.
+             */
+            if (__wt_backup_target_uri_search(session, metadata_key) == false) {
+                if (key->size > max_len)
+                    max_len = key->size;
+                WT_ERR(__wt_realloc_def(session, &allocated_name, slot + 2, &partial_backup_names));
+                p = &partial_backup_names[slot];
+                p[0] = p[1] = NULL;
+
+                WT_ERR(
+                  __wt_strndup(session, (char *)key->data, key->size, &partial_backup_names[slot]));
+                slot++;
+            }
+
+        /*
+         * In the case of partial backup restore, add the entry to the metadata even if the table
+         * entry doesn't exist so that we can correctly drop all related entries via the schema code
+         * later.
+         */
         WT_ERR(__wt_metadata_update(session, key->data, value->data));
     }
 
-    F_SET(S2C(session), WT_CONN_WAS_BACKUP);
+    F_SET(conn, WT_CONN_WAS_BACKUP);
+    if (F_ISSET(conn, WT_CONN_BACKUP_PARTIAL_RESTORE) && partial_backup_names != NULL) {
+        WT_ERR(__wt_calloc_def(session, slot + 1, &conn->partial_backup_remove_ids));
+        file_len = strlen("file:") + max_len + strlen(".wt") + 1;
+        /*
+         * Parse through the partial backup list and attempt to clean up all metadata references
+         * relating to the file. To do so, perform a schema drop operation on the table to cleanly
+         * remove all linked references. At the same time generate a list of btree ids to be used in
+         * recovery to truncate all the history store records.
+         */
+        for (slot = 0; partial_backup_names[slot] != NULL; ++slot) {
+            tablename = partial_backup_names[slot];
+            WT_PREFIX_SKIP_REQUIRED(session, tablename, "table:");
```

Comment:
Since we provide the assert that there should be no WT table in this list, I believe we don't need to do that here anymore. Do you agree @sueloverso ?